localize Categories In Home Screen And Media Details

M-Abdelnabi opened the pull request:
localize categories in Slider and Filter section of upcoming movies in home screen
localize categories in movie and tv-show in media details screen

M-Abdelnabi committed:
Refactor: Remove display name from Category enum

The `displayName` property has been removed from the `Category` enum in the Media domain. Each enum constant now only represents the category name itself.

M-Abdelnabi committed:
Feat: Add CategoryHomeUi enum for home feature display names

This commit introduces a new enum class `CategoryHomeUi` within the home UI feature. This enum maps category types to their corresponding display string resource IDs, allowing for localized category names in the user interface.

M-Abdelnabi committed:
Refactor: Update strings.xml and strings.xml (ar) to include genre names

M-Abdelnabi committed:
Refactor: Update CategoryNameMapper to use resource IDs for display names

The `CategoryNameMapper.kt` has been updated to use Android resource IDs for category display names instead of string literals.
This change introduces a new `CategoryHomeUi` enum to map domain `Category` enum values to their corresponding UI representations and string resource IDs.

The `toDisplayName` function now returns an `Int` (resource ID) instead of a `String`.
The function to convert a display name back to a `Category` enum (previously `String.toGenerEnum()`) has been updated to `Int.toCategory()` and now accepts a resource ID.

M-Abdelnabi committed:
Refactor: Update category and genre types from String to Int

The `categories` field in `AppSliderItem`, `HomeScreenUIState.MediaUi`, and the `genres` parameter in `DescriptionSection` and `GenreChipRow` have been changed from `List<String>` to `List<Int>`.

In `GenreChipRow`, `stringResource` is now used to get the genre title from the integer ID.
Preview data in `AppSlider` and `DescriptionSection` has been updated to reflect this change, using integer lists for categories/genres.

M-Abdelnabi committed:
Refactor: Update category display and mapping in home feature

The category display in `HomeScreen.kt` and `HomeSlider.kt` now uses string resources for localization. `HomeMapper.kt` was updated to correctly map categories. Test data in `HomeSection.kt` was also adjusted to reflect these changes.

M-Abdelnabi committed:
Refactor: Remove CategoryUiState from SearchUiState.kt

The data class `CategoryUiState` has been removed from `SearchUiState.kt` as it is no longer used. The import for `CategoryUiState` has also been removed from `SearchScreenInteractionListener.kt`.

M-Abdelnabi committed:
Feat: Add category strings for media details

This commit introduces new string resources for various media categories in both English and Arabic. These strings will be used to display localized category names within the media details feature.

M-Abdelnabi committed:
Feat: Add category strings for media details

This commit introduces new string resources for various media categories in both English and Arabic. These strings will be used to display localized category names within the media details feature.

M-Abdelnabi committed:
Feat: Add CategoryNameMapper to map domain Category to UI Category

This commit introduces `CategoryNameMapper.kt`.
This mapper includes:
- An extension function `toUiCategory()` to convert `Category` (domain) to `CategoryMediaUi` (UI).
- An extension function `toDisplayName()` that returns the display name resource ID for a `Category`.

M-Abdelnabi committed:
Refactor: Update genre type from String to Int in UI states and mappers

The `genres` field in `MovieUiState` and `TvShowUiState` has been changed from `List<String>` to `List<Int>`.
The `UiMapper` has been updated to use `it.toDisplayName()` when mapping categories to genres, reflecting this change.

M-Abdelnabi committed:
Refactor: Update HomeScreenViewModelTest to use Category enum and R.string resources

The `HomeScreenViewModelTest` has been updated to use the `Category` enum instead of string literals for categories. Additionally, category names in `MediaUiState` instances are now referenced using `R.string` resources.

The import for `toGenerEnum` was changed to `toCategory`.

These changes improve type safety and maintainability by utilizing enums and centralizing string resources.

M-Abdelnabi committed:
Refactor: Update ContinueWatchingViewModelTest to use string resources for categories

The categories in ContinueWatchingViewModelTest.kt have been updated to use string resources (R.string.category_drama and R.string.category_action) instead of hardcoded strings. This change improves localization and maintainability of the test data.

M-Abdelnabi committed:
Refactor: Update TopRatingMoviesViewModelTest.kt to use integer category IDs

The `TopRatingMoviesViewModelTest.kt` has been updated to use integer category IDs (CATEGORY_DRAMA and CATEGORY_COMEDY) instead of string literals for genre representation in the `fakeTopRatedList`. This change aligns the test data with a more structured approach to category management.

Joseph-Sameh-0 commented on feature/home/homeUi/src/main/java/com/feature/home/homeUi/screen/home/components/AppSlider.kt:
```diff
@@ -155,7 +155,7 @@ fun SliderPreview() {
                 title = "Sample Movie 1",
                 type = SliderMediaTypeUi.Movie,
-                categories = listOf("Action", "Adventure"),
+                categories = listOf(1, 2),
```
I think the int values here will crash

mohammed-akkad replied:
you can use annotation @StringRes

M-Abdelnabi replied:
This is just aPreview and it doesn't crash but i will edit it okay👍

mohammed-akkad commented on domain/media/src/main/java/com/paris_2/domain/media/entity/Category.kt:
```diff
@@ -1,33 +1,32 @@
+    MUSIC,
+    MYSTERY,
+    ROMANCE,
+    SCIENCE_FICTION,
```
you can use PascalCase instead of upper snack case
like this
ScienceFicition

M-Abdelnabi replied:
i think the preferred style is "UPPER_SNAKE_CASE" for enum constants

mohammed-akkad replied:
when we use jetpack compose recommend with it is pascal case

mohammed-akkad commented on feature/home/homeUi/src/main/java/com/feature/home/homeUi/mapper/CategoryHomeUi.kt:
```diff
@@ -0,0 +1,34 @@
+    MUSIC(R.string.category_music),
+    MYSTERY(R.string.category_mystery),
+    ROMANCE(R.string.category_romance),
+    SCIENCE_FICTION(R.string.category_science_fiction),
```
and here

mohammed-akkad commented on feature/mediaDetails/mediaDetailsUi/src/main/java/com/feature/mediaDetails/mediaDetailsUi/ui/mapper/CategoryMediaUi.kt:
```diff
@@ -0,0 +1,34 @@
+    MYSTERY(R.string.category_mystery),
+    ROMANCE(R.string.category_romance),
+    SCIENCE_FICTION(R.string.category_science_fiction),
+    TV_MOVIE(R.string.category_tv_movie),
```
and also here use pascal case

M-Abdelnabi committed:
Refactor: Update AppSlider preview to use string resources for categories

The preview data in `AppSlider.kt` has been updated to use string resources (e.g., `R.string.category_action`) for media categories instead of hardcoded integer values. This improves localization and maintainability.

M-Abdelnabi committed:
convert UPPER_SNAKE_CASE to PascalCase

M-Abdelnabi committed:
convert UPPER_SNAKE_CASE to PascalCase

M-Abdelnabi committed:
convert UPPER_SNAKE_CASE to PascalCase

M-Abdelnabi committed:
convert UPPER_SNAKE_CASE to PascalCase
